homeassistant/components/onewire/sensor.py

```diff
@@ -176,19 +203,34 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
                 )
 
     # We have a raw GPIO ow sensor on a Pi
-    elif base_dir == DEFAULT_SYSBUS_MOUNT_DIR:
-        _LOGGER.debug("Initializing using SysBus %s", base_dir)
-        for device_family in DEVICE_SENSORS:
-            for device_folder in glob(os.path.join(base_dir, f"{device_family}[.-]*")):
-                sensor_id = os.path.split(device_folder)[1]
-                device_file = os.path.join(device_folder, "w1_slave")
-                devs.append(
-                    OneWireDirect(
-                        device_names.get(sensor_id, sensor_id),
-                        device_file,
-                        "temperature",
-                    )
+    elif conf_type == CONF_TYPE_SYSBUS:
+        _LOGGER.debug("Initializing using SysBus")
+        for p1sensor in Pi1Wire().find_all_sensors():
+            family = p1sensor.mac_address[:2]
+            sensor_id = "{}-{}".format(family, p1sensor.mac_address[2:])
+            if family not in DEVICE_SUPPORT_SYSBUS:
```

Comment:
Why do we need this check?

Reply:
This is to prevent errors if more complex devices are plugged into the 1-wire bus as only simple temperature sensors are supported over SysBus.